src/xtgeo/xyz/points.py

```diff
@@ -158,11 +571,87 @@ def _random(self, nrandom=10):
+    @deprecation.deprecated(
+        deprecated_in="2.16",
+        removed_in="4.0",
+        current_version=xtgeo.version,
+        details="Use xtgeo.points_from_file() instead",
+    )
     def from_file(self, pfile, fformat="xyz"):
-        super().from_file(pfile, fformat=fformat)
+        self._reset(**_file_importer(pfile, fformat))
+
+    @deprecation.deprecated(
+        deprecated_in="2.16",
+        removed_in="4.0",
+        current_version=xtgeo.version,
+        details="Use xtgeo.points_from_roxar() instead.",
+    )
+    def from_roxar(
+        self,
+        project: Union[str, Any],
+        name: str,
+        category: str,
+        stype: str = "horizons",
+        realisation: int = 0,
+        attributes: bool = False,
+    ):  # pragma: no cover
+        """Load a points/polygons item from a Roxar RMS project (deprecated).
+
+        The import from the RMS project can be done either within the project
+        or outside the project.
+
+        Note that the preferred shortform for (use polygons as example)::
+
+          import xtgeo
+          mypoly = xtgeo.xyz.Polygons()
+          mypoly.from_roxar(project, 'TopAare', 'DepthPolys')
+
+        is now::
+
+          import xtgeo
+          mysurf = xtgeo.polygons_from_roxar(project, 'TopAare', 'DepthPolys')
+
+        Note also that horizon/zone/faults name and category must exists
+        in advance, otherwise an Exception will be raised.
+
+        Args:
+            project (str or special): Name of project (as folder) if
+                outside RMS, og just use the magic project word if within RMS.
+            name (str): Name of polygons item
+            category (str): For horizons/zones/faults: for example 'DL_depth'
+                or use a folder notation on clipboard.
+
+            stype (str): RMS folder type, 'horizons' (default) or 'zones' etc!
+            realisation (int): Realisation number, default is 0
+            attributes (bool): If True, attributes will be preserved (from RMS 11)
+
+        Returns:
+            Object instance updated
+
+        Raises:
+            ValueError: Various types of invalid inputs.
 
-        self._df.dropna(inplace=True)
+        .. deprecated:: 2.16
+           Use xtgeo.points_from_roxar() or xtgeo.polygons_from_roxar()
+        """
+        self._reset(
+            **_roxar_importer(
+                project,
+                name,
+                category,
+                stype,
+                realisation,
+                attributes,
+            )
+        )
 
+    @deprecation.deprecated(
+        deprecated_in="2.16",
+        removed_in="4.0",
+        current_version=xtgeo.version,
+        details="Use xtgeo.Points(xname=east, yname=north, zname=tvdmsl,"
+        " attributes=attributes) instead",
+    )
     def from_dataframe(self, dfr, east="X", north="Y", tvdmsl="Z", attributes=None):
         """Import points/polygons from existing Pandas dataframe.
```

Comment:
I cannot see how you would use the constructor directly. Example:
>>> df = pd.DataFrame({"Y": [1,2], "Z": [4,4], "A":[99, 99]})
>>> poi = Points()
>>> poi.from_dataframe(df, east="A", north="Y", tvdmsl="Z")   <<< note order of columns is changed
>>> poi._df
   X_UTME  Y_UTMN  Z_TVDSS
0      99       1        4
1      99       2        4

the alternative:
```
>>> poi2 = xtgeo.Points(df, xname="A", yname="Y", zname="Z")
>>> poi2._df
   Y  Z   A
0  1  4  99
1  2  4  99
```

Reply:
The only difference I see is a different order of columns and different value of `xname`, `yname` and `zname`. I believe that is not a problem for Points, and if it is then the setter, and init parameter of those properties is probably also problematic. The user is free to rename those columns themselves before passing it to the constructor.

 Is the order of columns significant anywhere? Then the constructor should probably normalize that.

Reply:
Yes the order of columns is important (required). Specifically; this method was designed to "inherit" selected columns (could be column no. 9, 2, and 7) in any input df with arbitrary number of columns, and use only those 3 (in correct order) to design a new Point() dataframe. Hence I would think the class method `points_from_dataframe()` from my PR is required.

Reply:
Then I think the `__init__` should reorder the columns or raise an error if it is not in the correct order. We can change the suggested replacement code to:
```python
Points(
  values=df[[xname, yname, tvdmsl]],
  xname=east,
  yname=north,
  zname=tvdmsl,
  attributes=attributes
)
```

Reply:
Yes, just tested it. May work yes :+1:

Reply:
Still a problem here:
```
p = xtgeo.Points(values=df[["F", "A", "Z"]], xname="X_UTME", yname="A", zname="Z")
>>> p
Points(values=       F    A    Z
0      0    0    0
1      1    1    1
2      2    2    2
3      3    3    3
4      4    4    4
..   ...  ...  ...
995  995  995  995
996  996  996  996
997  997  997  997
998  998  998  998
999  999  999  999

[1000 rows x 3 columns], xname=X_UTME, yname=A, zname=Z, attributes=None)

```

Reply:
Nice